### ISSUE TITLE
Errors occuring from new Mangadex@Home update

### ISSUE THREAD

**Issue by PythonCoderAS** [frozenpandaman/mangadex-dl#15]
It seems that using Mandadex@Home leads to all sorts of errors, and it seems to be on for guests by default. 

The first error has a simple patch:

```                                                                                                            
Traceback (most recent call last):                                                                                                                                                                                                          
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 157, in _new_conn                                                                                                                                               
    (self._dns_host, self.port), self.timeout, **extra_kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 61, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 752, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 994, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 300, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 169, in _new_conn
    self, "Failed to establish a new connection: %s" % e
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x1070d1590>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 436, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='mangadex.orghttps', port=443): Max retries exceeded with url: //4d00v29c8getr.rv0zp78t4x6km.mangadex.network:44300/data/e6ab60ca25e7102869bf1c599aabd4d9/g1.jpg (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x1070d1590>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "mangadex-dl.py", line 169, in <module>
    dl(manga_id, lang_code, url[-1])
  File "mangadex-dl.py", line 137, in dl
    r = scraper.get(url)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/cloudscraper/__init__.py", line 204, in request
    super(CloudScraper, self).request(method, url, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='mangadex.orghttps', port=443): Max retries exceeded with url: //4d00v29c8getr.rv0zp78t4x6km.mangadex.network:44300/data/e6ab60ca25e7102869bf1c599aabd4d9/g1.jpg (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x1070d1590>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
```

It seems that the returned URL is `https://mangadex.orghttps://4d00v29c8getr.rv0zp78t4x6km.mangadex.network:44300/data/e6ab60ca25e7102869bf1c599aabd4d9/g1.jpg`, so a simple way to fix this is to replace `https://mangadex.orghttps`, to `https://`. 

Another issue comes from SSL. 

Here's the issue: 
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 994, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 360, in connect
    ssl_context=context,
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 370, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 423, in wrap_socket
    session=session
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 870, in _create
    self.do_handshake()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 1139, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 436, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hd5ze167xfjda.za7a56fpsknn8.mangadex.network', port=443): Max retries exceeded with url: /data/289c5d98b80fd46f3458d8da7a15910a/v1.jpg (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "mangadex-dl.py", line 170, in <module>
    dl(manga_id, lang_code, url[-1])
  File "mangadex-dl.py", line 138, in dl
    r = scraper.get(url)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/cloudscraper/__init__.py", line 204, in request
    super(CloudScraper, self).request(method, url, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='hd5ze167xfjda.za7a56fpsknn8.mangadex.network', port=443): Max retries exceeded with url: /data/289c5d98b80fd46f3458d8da7a15910a/v1.jpg (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)')))
```

I have no clue how to fix this one.

**Comment by PythonCoderAS** [comment]
So it seems my solution for the second error is to wrap the entire thing in a try/except loop so that if any SSL errors occur the script idles for 5 seconds and requests another host. It seems that a different host may be returned each time you request chapter images.

**Comment by frozenpandaman** [comment]
Is there any announcement post about what "MandaDex@Home" is, or with more information, besides [this page](https://mangadex.org/md_at_home/info) I found via Google? Haven't heard of it…

**Comment by PythonCoderAS** [comment]
@frozenpandaman https://mangadex.org/thread/262074

**Comment by frozenpandaman** [comment]
Thanks! This was a super simple fix, sorry for not getting around to it before. Fixed in 6947854 / v0.2.5.
